pkg/middleware/logging.go

```diff
@@ -0,0 +1,37 @@
+// Copyright KubeArchive Authors
+// SPDX-License-Identifier: Apache-2.0
+package middleware
+
+import (
+	"log/slog"
+	"time"
+
+	"github.com/gin-gonic/gin"
+)
+
+func Logger(c *gin.Context) {
+	start := time.Now()
+	c.Next()
+	end := time.Now()
+
+	latency := end.Sub(start)
+	if latency > time.Minute { // Avoid very large latency number string
+		latency = latency.Truncate(time.Second)
+	}
+
+	path := c.Request.URL.Path
+	query := c.Request.URL.RawQuery
+	if query != "" {
+		path += "?" + query
+	}
+
+	slog.Info(
+		"Served request",
+		"method", c.Request.Method,
+		"status", c.Writer.Status(),
+		"latency", latency,
+		"client", c.ClientIP(),
+		"path", path,
+		"errors", c.Errors.ByType(1).String(),
```

Comment:
Should we use `Info` method if the served request response is >= 500 with errors?

Maybe we can do something like:

```suggestion
        var logLevel int
        if c.Writer.Status() >= 500 {
		logLevel = slog.LevelError
	} else {
		logLevel = slog.LevelInfo
	}
       
	slog.Log(
	        c, logLevel,
		"Served request",
		"method", c.Request.Method,
		"status", c.Writer.Status(),
		"latency", latency,
		"client", c.ClientIP(),
		"path", path,
		"errors", c.Errors.ByType(1).String(),
```

Reply:
I think it should be the function that returns the 500 that logs at the error level. This middleware is just "neutral" information.